### PR TITLE
Eventhub with Cleanup Policy

### PR DESCRIPTION
**Pull request by schaudhari6254888**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/eventhub/Azure.ResourceManager.EventHubs/src/Custom/EventHubData.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+using Azure.ResourceManager.Models;
+
+namespace Azure.ResourceManager.EventHubs
+{
+    /// <summary> A class representing the EventHubsNamespace data model. </summary>
+    public partial class EventHubData : ResourceData
+    {
+        [Obsolete("This property is obsolete and will be removed in a future release", false)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public long? MessageRetentionInDays { get; set; }
```
Comment (Yao725) [inline sdk/eventhub/Azure.ResourceManager.EventHubs/src/Custom/EventHubData.cs:20]: We should also keep the property description: `/// <summary> Number of days to retain the events for this Event Hub, value should be 1 to 7 days. </summary>`

Comment (schaudhari6254888) [reply sdk/eventhub/Azure.ResourceManager.EventHubs/src/Custom/EventHubData.cs:21]: @Yao725 I have added the summary.
Marked as resolved.

File: sdk/eventhub/Azure.ResourceManager.EventHubs/src/Generated/Models/EventHubsNetworkSecurityPerimeterConfiguration.cs [hunk 2]
```diff
@@ -12,11 +12,10 @@
 namespace Azure.ResourceManager.EventHubs.Models
 {
     /// <summary> Network Security Perimeter related configurations of a given namespace. </summary>
-    public partial class EventHubsNetworkSecurityPerimeterConfiguration : TrackedResourceData
+    public partial class EventHubsNetworkSecurityPerimeterConfiguration : ResourceData
```
Comment (m-nash) [inline sdk/eventhub/Azure.ResourceManager.EventHubs/src/Generated/Models/EventHubsNetworkSecurityPerimeterConfiguration.cs:15]: This is a breaking change https://apiview.dev/Assemblies/Review/55488f1a7fb94d18a477b5c98fa608cf/81cff07e8f6f4371aa80f8e7233a17ad?diffRevisionId=78fe242374404a77944c32ac310ab8ac&doc=False&diffOnly=True#895f0ee6a5164dd6bf0b87fdea2c65cf

**Comment by damodaravadhani** [comment]
Hey @Yao725, the build points to the obsolete MessageRetentionInDays. Should we suppress the build error or have I made a mistake here ?

**Comment by Yao725** [comment]
> Hey @Yao725 Yao Kou FTE, the build points to the obsolete MessageRetentionInDays. Should we suppress the build error or have I made a mistake here ?

@damodaravadhani, in my understanding, this property `MessageRetentionInDays` should be removed in the swagger as you said, so I suggest you adding the customized code to avoid breaking change and build failure. However, it looks like this property still exists, so you should remove the customization and just let the generator generate the code. And I'm also wondering if I misunderstood something here.

**Comment by damodaravadhani** [comment]
@Yao725 , as far as I can see, the generation did remove the property after rendering it obsolete, not really sure what we need to do next. Do you have any suggestions ?

File: sdk/eventhub/Azure.ResourceManager.EventHubs/tests/Samples/ChangeLog.cs [hunk 3]
```diff
@@ -43,7 +43,7 @@ public async Task ChangelogSample()
             EventHubCollection eventHubCollection = eventHubNamespace.GetEventHubs();
             EventHubData eventHubData = new EventHubData()
             {
-                MessageRetentionInDays = 4,
+                // MessageRetentionInDays = 4,
```
Comment (ArthurMa1978) [inline sdk/eventhub/Azure.ResourceManager.EventHubs/tests/Samples/ChangeLog.cs:46]: remove this line

File: sdk/eventhub/Azure.ResourceManager.EventHubs/api/Azure.ResourceManager.EventHubs.netstandard2.0.cs [hunk 4]
```diff
@@ -794,9 +815,9 @@ public partial class EventHubsNetworkSecurityPerimeter
         public Azure.Core.AzureLocation? Location { get { throw null; } }
         public string PerimeterGuid { get { throw null; } }
     }
-    public partial class EventHubsNetworkSecurityPerimeterConfiguration : Azure.ResourceManager.Models.TrackedResourceData
+    public partial class EventHubsNetworkSecurityPerimeterConfiguration : Azure.ResourceManager.Models.ResourceData
```
Comment (ArthurMa1978) [inline sdk/eventhub/Azure.ResourceManager.EventHubs/api/Azure.ResourceManager.EventHubs.netstandard2.0.cs:818]: This is a breaking change, why move the `location` out? What's the impact if move the `location` back as required property?

Comment (schaudhari6254888) [reply sdk/eventhub/Azure.ResourceManager.EventHubs/api/Azure.ResourceManager.EventHubs.netstandard2.0.cs:818]: Sorry for inconvenience. It was swagger problem. Now I have fixed the issue.

**Review by ArthurMa1978 — APPROVED** [review]

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**